source/common/stats/grpc_metrics_service_impl.h

```diff
@@ -0,0 +1,167 @@
+#pragma once
```

Comment:
Side note: Given how much of this code was mostly copied from my access log implementation, I feel like there could be a base class. All of this code is basically going to get copied again for the trace service implementation also. I would just add a TODO somewhere to look into converging this code and the access log code into a common base class.

Reply:
+1